src/CarbonAware.LocationSources/src/LocationSource.cs

```diff
@@ -32,100 +33,63 @@ public LocationSource(ILogger<LocationSource> logger, IOptionsMonitor<LocationDa
     {
         _logger = logger ?? throw new ArgumentNullException(nameof(logger));
         _configurationMonitor = monitor ?? throw new ArgumentNullException(nameof(monitor));
+        _namedGeopositions = new Dictionary<string, NamedGeoposition>();
     }
 
     public async Task<Location> ToGeopositionLocationAsync(Location location)
     {
-        switch (location.LocationType)
-        {
-            case LocationType.Geoposition:
-            {
-                return await Task.FromResult(location);
-            }
-            case LocationType.CloudProvider: 
-            {
-                var geoPositionLocation = await GetGeoPositionLocationOrThrowAsync(location);
-                return geoPositionLocation;
-            }
-        }
-        throw new LocationConversionException($"Location '{ location.LocationType }' cannot be converted to Geoposition. ");
-    }
-
-    private async Task<Location> GetGeoPositionLocationOrThrowAsync(Location location)
-    {
-        await LoadRegionsFromFileIfNotPresentAsync();
+        await LoadLocationFromFileIfNotPresentAsync();
 
-        var regionName = location.RegionName ?? string.Empty;
-        if (!_namedGeopositions!.ContainsKey(regionName))
+        var regionName = location.Name ?? string.Empty;
```

Comment:
Can we scrub this file for the "region" variables and messages?

```suggestion
        var name = location.Name ?? string.Empty;
```